Update best practice of writing selector (#612)
Azure/communication-ui-library: docs/architecture/WritingSelectors.md

A selector is a function built on top of reselect, it takes a list of functions as parameter and memoize their return value to decide if the second function provided in parameter need to be re-calculated, there are several principal we can follow when writing a selector for getting better perf:

> ### 1. Parameters for select must be as minimun as possible

Reviewer: selector? (and minimun -> minimum)
```suggestion
### 1. Parameters for selector must be as minimum as possible
```

> The first array parameter of createSelector() defines an parameters array for memoizing the return value of selector. No extra data in parameters should be memoize when designing what parameters to be memoized,

Reviewer: nits:
```suggestion
The first array parameter of createSelector() defines a parameter array for memoizing the return value of the selector. Only include necessary parameters in the array when designing what parameters to use for memoizing.
```

Don't:
```typescript
// state = {call:{ callId: UserId, displayname: string, callParticipants: Participants }}

> const getCall = (state) = > state.call;

Reviewer: nit: remove space in code sample
```suggestion
const getCall = (state) => state.call;
```

const callInfoSelector = createSelector([getCall], (call) => ({id: call.id, name: call.displayname }));
```

> In this _Don't_ example, the memoize dependencies array includes the whole call object, which will trigger a lot of undesired re-rendering when new participant join(which will update whole call object), and this is causing a perf regression.

Reviewer: nits:
```suggestion
In this _Don't_ example, the memoize dependency array includes the whole call object. This will trigger a lot of undesired re-rendering. For example, every time new participants join this will trigger an undesired re-render as when a participant joins the whole call object is updated.
```
